### PR TITLE
Update hypershift deployment crd

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>
issue reference: https://github.com/stolostron/backlog/issues/22689

This crd is synced from https://github.com/stolostron/hypershift-deployment-controller/blob/main/config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml , we need `replicas` field to config the desired number of nodes the nodepool should maintain.

### CODE REVIEW

File: pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml [hunk 1]
```diff
@@ -7497,6 +1917,11 @@ spec:
                           required:
                           - image
                           type: object
+                        replicas:
```
Comment (zhujian7) [inline pkg/templates/crds/hypershift/cluster.open-cluster-management.io_hypershiftdeployments.yaml:1920]: we need this to config the desired number of nodes

**Comment by zhujian7** [comment]
cc @stolostron/acm-hypershift @JakobGray

**Comment by JakobGray** [comment]
This says it's removing over 5000 lines. Is that correct? Is this PR still being updated?

**Comment by zhujian7** [comment]
@JakobGray I just copied it from [hypershift-deployment-controller repo](https://github.com/stolostron/hypershift-deployment-controller/blob/main/config/crd/cluster.open-cluster-management.io_hypershiftdeployments.yaml) which is the one matches the code, so I think it is right.
And I applied this CRD in my environment, it works fine.
cc @jnpacker @nelsonjean

**Comment by jnpacker** [comment]
@zhujian7 can you add to this comment, but we took a look and this aligns us with what is expected from the Hypershift commit SHA that's being used.

The big change is that Kubevirt hanged their CRD references.  We applied the CRD and it is not a concern for us. @rokej ran the Kubevirt test about 2x weeks back, we will re-run those tests to verify, but the CRD change for the kubevirt change aligns to hypershift-operator image SHA.

Again, we feel the risk is minimal, as the CRD applies and the changes are 98% in a dev preview portion of the code.

It's so big a change because we incorporate the Hypershift HostedClusterSpec and NodePoolSpec in our CRDs.

Again the risk after review is LOW.

**Comment by JakobGray** [comment]
@zkayyali812 Are we getting this CRD from somewhere via automation or is it manually updated?

**Comment by zhujian7** [comment]
> Are we getting this CRD from somewhere via automation or is it manually updated?

it is generated by `make manifests` for the hypershift-deployment-controller repo.
```
╰─# make manifests
/home/go/src/github.com/stolostron/hypershift-deployment-controller/bin/controller-gen rbac:roleName=hypershfit-deployment-controller crd paths="./..." output:crd:artifacts:config=config/crd
```

**Review by JakobGray — APPROVED** [review]

**Pull request merged**